docker-compose.yml

```diff
@@ -0,0 +1,39 @@
+version: "3.4"
+
+volumes:
+  mongodb_data_container:
+
+services:
+  nightscout:
+    image: "skalahonza/nightscout"
+    build:
+      context: .
+      dockerfile: Dockerfile    
+    environment:
+      API_SECRET: "${API_SECRET:-123456789abc}"
+      MONGO_CONNECTION: 'mongodb://${MONGODB_USERNAME:-nightscout}:${MONGODB_PASSWORD:-secret}@database/nightscout?retryWrites=true&w=majority&authSource=admin'
+      MONGO_COLLECTION: 'entries'
+      INSECURE_USE_HTTP: "true"
+      SECURE_HSTS_HEADER: "false"
+      PORT: 1337
+      TIME_FORMAT: "24"
+      DISPLAY_UNITS: "mmol"
+      WAIT_HOSTS: "database:27017"
```

Comment:
This doesn't work for me. I had to use something like

`command: sh -c "sleep 5; node server.js"`

Reply:
Can you pls post the std out output in the comments? It works fine on my machine.

Reply:
Actually I am not using your branch, but my docker compose file is very similar. I've also added `WAIT_HOSTS`. I think it is a timing problem. As you can see in the startup sequence the nightscout-app is not waiting long enough. Although nightscout-app makes afterward a successful connection, Nightscout doesn't work. I didn't have this problem with the same docker-compose.yml file before upgrading to 14.0.7.

```
nightscout-db | 2020-11-27T16:01:00.060+0000 I CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=a6370c7791ae

...

nightscout-app | Error connecting to MongoDB: {"name":"MongoNetworkError"} - retrying in 3 sec
nightscout-app | ERROR CONNECTING TO MONGO Error: MongoDB connection failed! Double check the MONGODB_URI setting in Heroku.
nightscout-app |     at MongoClient.connect.then.catch.err (/opt/app/lib/storage/mongo-storage.js:75:30)
nightscout-app |     at process._tickCallback (internal/process/next_tick.js:68:7)
nightscout-app | Mongo Storage system ready

...

nightscout-db | 2020-11-27T16:01:01.177+0000 I NETWORK  [initandlisten] waiting for connections on port 27017

...

nightscout-app | Successfully established a connected to MongoDB
nightscout-app | Mongo user role seems ok: [ { role: 'root', db: 'admin' } ]
nightscout-app | Mongo Storage system ready

```

Reply:
WAIT_HOSTS is not a built in functionality, I had to add support for it in the Dockerfile
```
## Add the wait script to the image
ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.7.3/wait /wait
RUN chmod +x /wait
```